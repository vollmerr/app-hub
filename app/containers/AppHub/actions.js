/*
 *
 * AppHub actions
 *
 */

import {
  CHANGE_MOBILE,
} from './constants';

export function changeMobile(isMobile) {
  return {
    isMobile,
    type: CHANGE_MOBILE,
  };
}
