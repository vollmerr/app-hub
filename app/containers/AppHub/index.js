import React from 'react';
import styled from 'styled-components';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { compose } from 'redux';
import { Fabric } from 'office-ui-fabric-react/lib/Fabric';

import Header from './Header';
import * as selectors from './selectors';


export const Wrapper = styled(Fabric) `
  height: 100%;
  overflow: hidden;
`;


export class AppHub extends React.PureComponent {
  render() {
    return (
      <Wrapper>
        <Header />
        {
          isAuthenticated ?
            <Router routes={routes} /> :
            <Loading pastDelay {...this.props.app} />
        }
      </Wrapper>
    );
  }
}

const mapStateToProps = createStructuredSelector({
  isAuthenticated: selectors.getIsAuthenticated,
});


export default compose(
  withSaga,
  withConnect,
)(AppHub);

// tests....
