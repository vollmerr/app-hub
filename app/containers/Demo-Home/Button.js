import React from 'react';

const Button = ({ onClick, disabled, text }) => ( // eslint-disable-line
  <button
    onClick={onClick}
    disabled={disabled}
    style={{ background: disabled ? '#666' : '#333', color: '#fff', padding: '10px 15px', cursor: 'pointer' }}
  >{text}</button>
);

export default Button;
