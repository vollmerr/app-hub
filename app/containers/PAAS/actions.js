import { createAction } from 'redux-actions';

import * as C from './constants';


export const getManagerDataRequest = createAction(C.GET_MANAGER_DATA_REQUEST);
export const getManagerDataSuccess = createAction(C.GET_MANAGER_DATA_SUCCESS);
export const getManagerDataFailure = createAction(C.GET_MANAGER_DATA_FAILURE);

export const getReportDataRequest = createAction(C.GET_REPORT_DATA_REQUEST);
export const getReportDataSuccess = createAction(C.GET_REPORT_DATA_SUCCESS);
export const getReportDataFailure = createAction(C.GET_REPORT_DATA_FAILURE);

export const updateUsersRequest = createAction(C.UPDATE_USERS_REQUEST);
export const updateUsersSuccess = createAction(C.UPDATE_USERS_SUCCESS);
export const updateUsersFailure = createAction(C.UPDATE_USERS_FAILURE);
