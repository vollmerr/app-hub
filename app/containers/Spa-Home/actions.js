/*
 *
 * SpaHome actions
 *
 */

import {
  EXAMPLE_DATA_REQUEST,
  EXAMPLE_DATA_SUCCESS,
  EXAMPLE_DATA_FAILURE,
} from './constants';

export function exampleDataRequest() {
  return {
    type: EXAMPLE_DATA_REQUEST,
  };
}

export function exampleDataSuccess(data) {
  return {
    data,
    type: EXAMPLE_DATA_SUCCESS,
  };
}

export function exampleDataFailure(error) {
  return {
    error,
    type: EXAMPLE_DATA_FAILURE,
  };
}
