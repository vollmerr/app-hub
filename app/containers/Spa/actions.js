import { createAction } from 'redux-actions';

import {
  INIT_DATA_REQUEST,
  INIT_DATA_SUCCESS,
  INIT_DATA_FAILURE,
  NEW_ACK_REQUEST,
  NEW_ACK_SUCCESS,
  NEW_ACK_FAILURE,
  DISABLE_ACK_REQUEST,
  DISABLE_ACK_SUCCESS,
  DISABLE_ACK_FAILURE,
  GET_RECIPIENTS_REQUEST,
  GET_RECIPIENTS_SUCCESS,
  GET_RECIPIENTS_FAILURE,
} from './constants';
// INITAL DATA
export const initDataRequest = createAction(INIT_DATA_REQUEST);
export const initDataSuccess = createAction(INIT_DATA_SUCCESS);
export const initDataFailure = createAction(INIT_DATA_FAILURE);
// NEW ACKNOWLEDGMENT
export const newAckRequest = createAction(NEW_ACK_REQUEST);
export const newAckSuccess = createAction(NEW_ACK_SUCCESS);
export const newAckFailure = createAction(NEW_ACK_FAILURE);
// DSIABLE ACKNOWLEDGMENT
export const disableAckRequest = createAction(DISABLE_ACK_REQUEST);
export const disableAckSuccess = createAction(DISABLE_ACK_SUCCESS);
export const disableAckFailure = createAction(DISABLE_ACK_FAILURE);
// GET LIST OF RECIPIENTS FOR EXISTSING ACK
export const getRecipientsRequest = createAction(GET_RECIPIENTS_REQUEST);
export const getRecipientsSuccess = createAction(GET_RECIPIENTS_SUCCESS);
export const getRecipientsFailure = createAction(GET_RECIPIENTS_FAILURE);
