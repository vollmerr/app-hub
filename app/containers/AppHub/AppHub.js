/**
 *
 * AppHub
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { compose } from 'redux';
// import AppHubHeader from 'containers/AppHub-Header';
import AppHubHeader from 'components/AppHub-Header';
import AppHubPanel from 'components/AppHub-Panel';
import theme from 'utils/theme';

import injectSaga from 'utils/injectSaga';
// import injectReducer from 'utils/injectReducer';
import {
  makeSelectIsMobile,
  makeSelectPanelIsOpen,
  makeSelectPanelSelected,
  makeSelectAppRoutes,
  makeSelectAppMeta,
  makeSelectUserSam,
} from './selectors';
// import reducer from './reducer';
import saga from './saga';
import { changeMobile, changePanelOpen, changePanelSelected, authUserRequest } from './actions';

export class AppHub extends React.Component { // eslint-disable-line react/prefer-stateless-function
  constructor(props) {
    super(props);
    this.state = {};
  }

  componentDidMount() {
    this.props.dispatch(authUserRequest());
    this.handleResize();
    window.addEventListener('resize', this.handleResize);
  }

  componentWillUnmount() {
    window.removeEventListener('resize', this.handleResize);
  }

  handleResize = () => {
    const { dispatch, isMobile } = this.props;
    const mobile = window.innerWidth <= theme.breakpoints.lg;

    if (mobile !== isMobile) {
      dispatch(changeMobile(mobile));
    }
  }

  handlePanelClick = (panel) => {
    const { dispatch, panelSelected, panelIsOpen } = this.props;
    const selected = panel || panelSelected; // errors if null, so set a default
    // click on same panel or overlay
    if (!panel || (panelSelected === panel && panelIsOpen)) {
      // close it
      dispatch(changePanelOpen(false));
    } else {
      // open the panel (don't dispatch unessecary actions if already open)
      if (!panelIsOpen) {
        dispatch(changePanelOpen(true));
      }
      dispatch(changePanelSelected(selected));
    }
  }

  render() {
    const { isMobile, panelSelected, panelIsOpen, appMeta, appRoutes, userName } = this.props;

    const headerProps = {
      isMobile,
      userName,
      panel: panelSelected,
      isOpen: panelIsOpen,
      appName: appMeta.name,
      appPath: appRoutes.find((route) => route.key === 'home').path,
      onClick: this.handlePanelClick,
    };

    const panelProps = {
      routes: appRoutes,
      panel: panelSelected,
      isOpen: panelIsOpen,
      onClick: this.handlePanelClick,
    };

    return (
      <div>
        <AppHubHeader {...headerProps} />
        <AppHubPanel {...panelProps} />
      </div>
    );
  }
}

AppHub.propTypes = {
  dispatch: PropTypes.func.isRequired,
  isMobile: PropTypes.bool.isRequired,
  panelIsOpen: PropTypes.bool.isRequired,
  panelSelected: PropTypes.string.isRequired,
  // appName: PropTypes.string,
  // appPath: PropTypes.string,
};

const mapStateToProps = createStructuredSelector({
  isMobile: makeSelectIsMobile(),
  panelIsOpen: makeSelectPanelIsOpen(),
  panelSelected: makeSelectPanelSelected(),
  appRoutes: makeSelectAppRoutes(),
  appMeta: makeSelectAppMeta(),
  userName: makeSelectUserSam(),
});

function mapDispatchToProps(dispatch) {
  return {
    dispatch,
  };
}

const withConnect = connect(mapStateToProps, mapDispatchToProps);

// const withReducer = injectReducer({ key: 'appHub', reducer });
const withSaga = injectSaga({ key: 'appHub', saga });

export default compose(
  // withReducer,
  withSaga,
  withConnect,
)(AppHub);
