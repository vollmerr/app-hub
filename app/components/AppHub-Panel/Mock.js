import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { DefaultButton } from 'office-ui-fabric-react/lib/Button';
import { Dropdown } from 'office-ui-fabric-react/lib/Dropdown';
import { TextField } from 'office-ui-fabric-react/lib/TextField';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import { authUser } from 'containers/AppHub/actions';
import { makeSelectUserSid } from 'containers/AppHub/selectors';
import theme from 'utils/theme';

import Wrapper from './Wrapper';


export const Field = styled.div`
  label {
    color: ${theme.white};
  }
`;


/**
 * Panel for mocking data utilities, such as changing user jwt.
 * (This should be in containers, but makes more sense to have with other panels)
 */
export class Mock extends React.PureComponent {
  handleClearToken = () => {
    localStorage.removeItem('id_token');
    window.location.reload();
  }

  handleClickUser = (user) => {
    const { onAuthUser, onClick } = this.props;

    if (user && user.key) {
      localStorage.setItem('id_token', user.key);
    } else {
      localStorage.removeItem('id_token');
    }
    onAuthUser();
    onClick();
    window.location.reload();
  }

  render() {
    const { userSid } = this.props;
    const jwts = MOCK.JWT; // eslint-disable-line

    return (
      jwts ?
        <Wrapper>
          <h2>Mock Options</h2>
          <hr />
          <DefaultButton
            text={'Clear Token'}
            onClick={this.handleClearToken}
          />
          <h3>Switch User</h3>
          <Field>
            <TextField
              disabled
              label={'Current User'}
              value={userSid}
            />
          </Field>
          {
            Object.keys(jwts).map((group) => (
              <Field key={group}>
                <Dropdown
                  label={`${group} Test Users`}
                  id={group}
                  options={jwts[group]}
                  onChanged={this.handleClickUser}
                />
              </Field>
            ))
          }
        </Wrapper> :
        null
    );
  }
}


const { func, string } = PropTypes;

Mock.propTypes = {
  onAuthUser: func.isRequired,
  userSid: string,
  onClick: func.isRequired,
};

const mapStateToProps = createStructuredSelector({
  userSid: makeSelectUserSid(),
});

export function mapDispatchToProps(dispatch) {
  return {
    onAuthUser: () => dispatch(authUser()),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(Mock);
