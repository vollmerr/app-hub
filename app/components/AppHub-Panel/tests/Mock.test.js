import React from 'react';
import { shallow } from 'enzyme';
import { DefaultButton } from 'office-ui-fabric-react/lib/Button';

import { testStyledComponent } from 'utils/testUtils';
import { authUser } from 'containers/AppHub/actions';

import { Mock, Field, mapDispatchToProps } from '../Mock';

testStyledComponent(Field);

const user1 = { name: 'test name 1', key: 'test key 1' };
const user2 = { name: 'test name 3', key: 'test key 3' };

global.MOCK = {
  JWT: {
    group1: [
      user1,
      { name: 'test name 2', key: 'test key 2' },
    ],
    group2: [
      user2,
      { name: 'test name 4', key: 'test key 4' },
    ],
  },
};

describe('<Mock />', () => {
  let wrapper;
  let onClick;
  let onAuthUser;
  beforeEach(() => {
    onClick = jest.fn();
    onAuthUser = jest.fn();
    wrapper = shallow(<Mock onClick={onClick} onAuthUser={onAuthUser} />);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a h2 header', () => {
    expect(wrapper.find('h2').length).toEqual(1);
  });

  it('should render a button for clearing the token from localStorage', () => {
    expect(wrapper.find(DefaultButton).length).toEqual(1);
  });

  it('should render the groups in their own group', () => {
    const groups = wrapper.findWhere((x) => x.prop('options'));
    // there are 2 groups
    expect(groups.length).toEqual(2);
    // user 1 in group 1, but not user2
    expect(groups.at(0).prop('options').includes(user1)).toEqual(true);
    expect(groups.at(0).prop('options').includes(user2)).toEqual(false);
    // user 2 in group 2, but not user1
    expect(groups.at(1).prop('options').includes(user1)).toEqual(false);
    expect(groups.at(1).prop('options').includes(user2)).toEqual(true);
  });

  describe('handleClearToken', () => {
    let instance;
    beforeEach(() => {
      global.localStorage.removeItem = jest.fn();
      window.location.reload = jest.fn();
      instance = wrapper.instance();
    });

    it('should exist', () => {
      expect(instance.handleClearToken).toBeDefined();
    });

    it('should remove the token from localstorage then refresh the window', () => {
      instance.handleClearToken();
      expect(global.localStorage.removeItem).toHaveBeenCalledWith('id_token');
      expect(window.location.reload).toHaveBeenCalled();
    });
  });

  describe('handleClickUser', () => {
    let instance;
    beforeEach(() => {
      global.localStorage = {
        setItem: jest.fn(),
        removeItem: jest.fn(),
      };
      instance = wrapper.instance();
    });

    it('should exist', () => {
      expect(instance.handleClickUser).toBeDefined();
    });

    it('should dispatch onAuthUserRequest when called', () => {
      instance.handleClickUser(user1);
      expect(onAuthUser).toHaveBeenCalled();
    });

    it('should update the local token if a user with a key is passed', () => {
      const setItem = global.localStorage.setItem;
      instance.handleClickUser(user1);
      expect(setItem).toHaveBeenCalled();
    });

    it('should clear the local token if no user or user key', () => {
      const removeItem = global.localStorage.removeItem;
      instance.handleClickUser();
      expect(removeItem).toHaveBeenCalled();
    });
  });

  describe('mapDispatchToProps', () => {
    let mappedDispatch;
    let dispatch;
    beforeEach(() => {
      dispatch = jest.fn();
      mappedDispatch = mapDispatchToProps(dispatch);
    });

    describe('onAuthUser', () => {
      it('should be injected', () => {
        expect(mappedDispatch.onAuthUser).toBeDefined();
      });

      it('should dispatch authUser when called', () => {
        mappedDispatch.onAuthUser();
        expect(dispatch).toHaveBeenCalledWith(authUser());
      });
    });
  });

  // KEEP LAST AS MODIFYING GLOBAL VARIABLE
  it('should not render if MOCK.JWT does not exist (production/dev)', () => {
    global.MOCK.JWT = undefined;
    wrapper = shallow(<Mock onClick={onClick} onAuthUser={onAuthUser} />);
    expect(wrapper.equals(null)).toEqual(true);
  });
});
