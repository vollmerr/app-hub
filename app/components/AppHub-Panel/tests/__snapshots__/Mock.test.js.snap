// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Mock /> should render correctly 1`] = `
<Wrapper__Wrapper>
  <h2>
    Mock Options
  </h2>
  <hr />
  <CustomizedDefaultButton
    onClick={[Function]}
    text="Clear Token"
  />
  <h3>
    Switch User
  </h3>
  <Mock__Field>
    <TextField
      autoAdjustHeight={false}
      borderless={false}
      deferredValidationTime={200}
      disabled={true}
      errorMessage=""
      label="Current User"
      multiline={false}
      onBeforeChange={[Function]}
      onChanged={[Function]}
      onGetErrorMessage={[Function]}
      onNotifyValidationResult={[Function]}
      resizable={true}
      underlined={false}
      validateOnFocusIn={false}
      validateOnFocusOut={false}
      validateOnLoad={true}
    />
  </Mock__Field>
  <Mock__Field
    key="group1"
  >
    <WithResponsiveMode
      id="group1"
      label="group1 Test Users"
      onChanged={[Function]}
      options={
        Array [
          Object {
            "key": "test key 1",
            "name": "test name 1",
          },
          Object {
            "key": "test key 2",
            "name": "test name 2",
          },
        ]
      }
    />
  </Mock__Field>
  <Mock__Field
    key="group2"
  >
    <WithResponsiveMode
      id="group2"
      label="group2 Test Users"
      onChanged={[Function]}
      options={
        Array [
          Object {
            "key": "test key 3",
            "name": "test name 3",
          },
          Object {
            "key": "test key 4",
            "name": "test name 4",
          },
        ]
      }
    />
  </Mock__Field>
</Wrapper__Wrapper>
`;

exports[`<Mock__Field /> should render correctly 1`] = `
.c0 label {
  color: #ffffff;
}

<div
  className="c0"
/>
`;
